### DISCUSSION TITLE
How pytest resolves import

### DISCUSSION THREAD

**Discussion by michaelx1999** [pytest-dev/pytest#12642] [Q&A]
Here is my folder structure

```
/root
  /app
    __init__.py
    /main.py
  /tests
    test_something.py (from app.main import ...)
```

When I run pytest inside /root or /root/tests, I get the error message "cannot import module app".

I change the folder structure to this and everything works fine.

```
/root
  /app
    __init__.py
    /main.py
  /tests
    __init__.py
    test_something.py (from app.main import ...)
```

1. How does adding this init file under tests resolves import error
2. How come pytest are able to resolves import correctly when I'm inside different directories? (/root, /root/tests)

**Comment by wimglenn** [comment]
I answered the same question already on stackoverflow, here:
https://stackoverflow.com/a/72368703/674039

**Comment by markshao** [comment]
you should run under root,otherwise add roo
t to pythonpath
